pkg/identity/certificated.go

```diff
@@ -82,6 +82,91 @@ func Certificated(idConfig *IdentityConfig, stopChan <-chan struct{}) (error, <-
 		return w.Save()
 	}
 
+	identityProvisioningRequest := func(idConfig *IdentityConfig, handler *identityHandler, forceInit bool) (err error, identity *InstanceIdentity, keyPem []byte) {
+		log.Infof("Mapped Athenz domain[%s], service[%s]", handler.Domain(), handler.Service())
+
+		identity, keyPem, err = handler.GetX509Cert(forceInit)
+
+		if err != nil {
+			log.Warnf("Error while requesting x509 certificate to identity provider: %s", err.Error())
+
+			if idConfig.CertSecret != "" && strings.Contains(idConfig.Backup, "write") {
+				log.Errorf("Failed to receive x509 certificate to update kubernetes secret[%s]: %s", idConfig.CertSecret, err.Error())
+				return
+			}
+		} else {
+			log.Infoln("Successfully received x509 certificate from identity provider")
+
+			if idConfig.CertSecret != "" && strings.Contains(idConfig.Backup, "write") {
+
+				log.Infof("Attempting to save x509 certificate to kubernetes secret[%s]...", idConfig.CertSecret)
+
+				err = handler.ApplyX509CertToSecret(identity, keyPem)
+				if err != nil {
+					log.Errorf("Failed to save x509 certificate to kubernetes secret: %s", err.Error())
+					return
+				}
+
+				log.Infof("Successfully saved x509 certificate to kubernetes secret")
+			} else {
+				log.Debugf("Skipping to save x509 certificate temporary backup to Kubernetes secret[%s]", idConfig.CertSecret)
+			}
+		}
+
+		return
+	}
+
+	roleCertProvisioningRequest := func(idConfig *IdentityConfig, handler *identityHandler) (roleCerts [](*RoleCertificate)) {
```

Comment:
Do you think we need this
`idConfig *IdentityConfig, handler *identityHandler`?

Seems like we don't as it is defined above anyway.

Reply:
@ctyano Does this apply to the `identityProvisioningRequest` too?
Will handle them all later after you review the logic part.

Reply:
@ctyano handled!